src/chromium/adding-third-party-crates/generating-gn-build-rules.md

```diff
@@ -0,0 +1,64 @@
+# Generating gn build rules
+
+A tool called `gnrt` converts from `third_party.toml` into the `BUILD.gn` rules
+required by our build system. It can also download crate source code.
+
+At the moment, this is a multi-step process - it may be streamlined in future.
+You'll need to mix:
+
+* generating build files
+* downloading missing crates
+
+and possibly run through the cycle a couple of times.
+
+To start, download the crate you want like this:
+
+```
+cd chromium/src
+vpython3 tools/crates/run_gnrt.py -- download CRATE_NAME CRATE_VERSION  --shipped yes --security-critical yes
+```
+
+**IMPORTANT:** although the `gnrt` tool is part of the Chromium source code,
+by running this command you will be downloading and running its dependencies
+from `crates.io`. See [the earlier section][1] discussing this security
+decision.
+
+In this case, you need to specify:
+* The crate name. Replace any underscores with hyphens.
```

Comment:
Maybe not very important, but does this also apply to a crate like https://crates.io/crates/mockall_derive which is truly named `mockall_derive` with an underscore?

Reply:
Good question, I'll try to find out.

Reply:
These instructions have now been deleted as the tooling has moved on.